tests/func/test_check_ignore.py

```diff
@@ -1,11 +1,31 @@
 import os
+from pathlib import Path
 
 import pytest
 
 from dvc.cli import main
 from dvc.ignore import DvcIgnore
 
 
+def _get_tmp_config_dir(tmp_path, level):
+    return tmp_path / ".." / (tmp_path.name + "_config_" + level)
+
+
+@pytest.fixture(autouse=True)
+def tmp_config_dir(mocker, tmp_path):
+    """
+    Fixture to prevent modifying/reading the actual global config
+    """
+
+    for level in ["global", "system"]:
+        os.makedirs(_get_tmp_config_dir(tmp_path, level))
+
+    def get_tmp_config_dir(level):
+        return str(_get_tmp_config_dir(tmp_path, level))
+
+    mocker.patch("dvc.config.Config.get_dir", side_effect=get_tmp_config_dir)
```

Comment:
This shouldn't be needed, we already isolate the environment in the `isolate` fixture from `tests/conftest.py`.

Reply:
So when I was writing the tests, the `.dvcignore` from my system dir (`/Library/Application Support/dvc/.dvcignore`) was being read in the tests, which is why I ended up mocking `Config.get_dir`.

Now that I look into it further, the `isolate` fixture does return an isolated directory for `Config.get_dir("global")` but does not seem to so for `Config.get_dir("system")` and returns `/Library/Application Support/dvc` instead.

I'll see if I can make the `isloate` fixture also isolate the system config directory.

Reply:
So it turns out that the `isolate` fixture in it's current is not completely ready to handle global/system dvc configs. The PR that added the this fixture mentions this fact in a comment: https://github.com/iterative/dvc/pull/6857#issuecomment-1007151300.

Tried to cover all cases in the following way:
https://github.com/iterative/dvc/pull/8280/files#diff-e52e4ddd58b7ef887ab03c04116e676f6280b824ab7469d5d3080e5cba4f2128

There were some situations where manipulating env vars was not helpful. In such cases, this PR mocks the functions from `appdirs` instead. 

Lmk what you think of this